packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -50,6 +55,25 @@ import {readContextDuringReconcilation} from './ReactFiberNewContext';
 let thenableState: ThenableState | null = null;
 let thenableIndexCounter: number = 0;
 
+function mergeDebugInfo(
+  outer: ReactDebugInfo | null,
+  inner: ReactDebugInfo | null | void,
+): ReactDebugInfo | null {
+  if (!__DEV__) {
+    return null;
+  }
+  if (inner == null) {
+    return outer;
+  } else if (outer === null) {
+    return inner;
+  } else {
+    // If we have two debugInfo, we need to create a new one. This makes the array no longer
+    // live so we'll miss any future updates if we received more so ideally we should always
+    // do this after both have fully resolved/unsuspended.
+    return outer.concat(inner);
```

Comment:
should we just let the debug info optionally be a tuple of debug infos that get flattened wherever they get turned into stacks? then we can write into each array indefinitely?

Reply:
Yea I considered that but it's also annoying given that this will almost never happen. Since the Flight renderer will have a single row for (almost?) all these cases.

Reply:
It is perhaps in conflict with my other point that DebugInfo might become per-instance objects of some sorts too.